src/storage/gcs/storage.c

```diff
@@ -98,6 +98,61 @@ struct StorageGcs
     HttpClient *authClient;                                         // Client to service auth requests
 };
 
+/***********************************************************************************************************************************
+Parse HTTP JSON response containing an authentication token and expiration
+
+Note that the function is intended to run directly in the caller's mem context and results will be placed in the caller's prior mem
+context.
+***********************************************************************************************************************************/
+typedef struct
+{
+    String *tokenType;
+    String *token;
+    time_t timeExpire;
+} StorageGcsAuthTokenResult;
+
+static StorageGcsAuthTokenResult
+storageGcsAuthToken(HttpRequest *request, time_t timeBegin)
+{
+    FUNCTION_TEST_BEGIN();
+        FUNCTION_TEST_PARAM(HTTP_REQUEST, request);
+        FUNCTION_TEST_PARAM(TIME, timeBegin);
+    FUNCTION_TEST_END();
```

Comment:
Some indentation issues here

Reply:
Not sure what you mean. Is it that you don't expect FUNCTION_TEST_PARAM() to be indented? That's conventions we used in these debug preambles to make them a bit more readable.

Reply:
Oh ... I missed that it was BEGIN, PARAM, PARAM, END ... I read all of them as the same macro. Never mind.